src/Microsoft.ML.HalLearners/OlsLinearRegression.cs

```diff
@@ -65,15 +64,57 @@ public sealed class Arguments : LearnerInputBaseWithWeight
         private static readonly TrainerInfo _info = new TrainerInfo(caching: false);
         public override TrainerInfo Info => _info;
 
-        public OlsLinearRegressionTrainer(IHostEnvironment env, Arguments args)
-            : base(env, LoadNameValue)
+        /// <summary>
+        /// Initializes a new instance of <see cref="OlsLinearRegressionTrainer"/>
+        /// </summary>
+        /// <param name="env">The private instance of <see cref="IHostEnvironment"/>.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="featureColumn">The name of the feature column.</param>
+        /// <param name="weightColumn">The name for the column containing the initial weight.</param>
+        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
+        public OlsLinearRegressionTrainer(IHostEnvironment env, string featureColumn, string labelColumn,
+            string weightColumn = null, Action<Arguments> advancedSettings = null)
+            : this(env, ArgsInit(featureColumn, labelColumn, weightColumn, advancedSettings))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="OlsLinearRegressionTrainer"/>
+        /// </summary>
+        internal OlsLinearRegressionTrainer(IHostEnvironment env, Arguments args)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(LoadNameValue), TrainerUtils.MakeR4VecFeature(args.FeatureColumn),
+                  TrainerUtils.MakeR4ScalarLabel(args.LabelColumn), TrainerUtils.MakeR4ScalarWeightColumn(args.WeightColumn))
         {
             Host.CheckValue(args, nameof(args));
             Host.CheckUserArg(args.L2Weight >= 0, nameof(args.L2Weight), "L2 regularization term cannot be negative");
             _l2Weight = args.L2Weight;
             _perParameterSignificance = args.PerParameterSignificance;
         }
 
+        private static Arguments ArgsInit(string featureColumn, string labelColumn,
+            string weightColumn, Action<Arguments> advancedSettings)
+        {
+            var args = new Arguments();
+
+            //apply the advanced args, if the user supplied any
```

Comment:
>//apply the advanced args, if the user supplied any [](start = 12, length = 51)

` // Comments need to be 1 space away from the comment symbol, begin with a capital letter and and with a period.